### ISSUE TITLE
Rule proposal: `no-bitwise-trunc`

### ISSUE THREAD

**Issue by papb** [sindresorhus/eslint-plugin-unicorn#827]
Prefer the [`Math.trunc`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/trunc) method for truncating numbers instead of the `| 0` hack which doesn't even work in some cases (such as `3000000000.1 | 0`, which gives `-1294967296`).

## Fail
```js
const bar = foo | 0;
```

## Pass
```js
const bar = Math.trunc(foo);
```

[Relevant SO question](https://stackoverflow.com/questions/7487977/using-bitwise-or-0-to-floor-a-number)

**Comment by sindresorhus** [comment]
Accepted. PR welcome.
